Sources/Rendering/Misc/SynchronizableRenderWindow/vtkObjectManager.js

```diff
@@ -261,20 +261,24 @@ function genericUpdater(instance, state, context) {
         .finally(context.end); // -> end(arrays)
     });
     context.start(); // -> start(arraysToBind)
-    Promise.all(promises)
-      .then(() => {
-        bindArrays(arraysToBind);
-      })
-      .catch((error) => {
-        console.error(
-          'Error in array handling for state',
-          JSON.stringify(state),
-          error
-        );
-      })
-      .finally(context.end); // -> end(arraysToBind)
+    dependencies.push(
+      ...promises
+        .then(() => {
```

Comment:
I'm surprised you can do `...promises.then()`

You can do: 
 1) bind arrays as soon as they have been fetched, but this is a new behavior though, 
 2) or bind them in the `Promises.all(dependencies).then()`. But it looses "generic" meaning of "dependencies.
 3) `dependencies.push(promises); dependencies.push(Promise.all(promises)...)` and then `return Promise.all(dependencies)`

I would go for Solution 3, but I would be curious to know what @jourdain would prefer

Reply:
This syntax is wrong indeed. 

Please read my initial comments, I gave you the exact code to write that would be correct. 

```
dependencies.push(...promises);
[...]
return Promise.all(dependencies);
```

Reply:
dependencies should only contains promises, no need to nest some Promise.all in them. I would rather keep it flat rather than having a useless hierarchy.

Reply:
With what you suggest Sebastien, you take the risk to have the returned promise being resolved BEFORE the arrays are bound. I guess it is fine but that means the application can't ask for a re-render when the returned promise is resolved (i.e. arrays may not yet be bound).

Reply:
Then just add a promise in your dependencies list that will only resolve once the bind happen...

Reply:
then it should be what Antoine did with his last [push](https://github.com/Kitware/vtk-js/pull/2025/commits/fbf4bc10ada68e2210a6617d37e3af600d818c63). But I guess he can remove the intermediate promises.

Reply:
Kind of, yes. With some small error above.

Reply:
In summary Antoine just need to remove `dependencies.push(promises);` as this is the only remaining thing that is wrong.